qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/AsymmetricNetworkPartitionIT.java

```diff
@@ -0,0 +1,287 @@
+
+  private ZeebeClient zeebeClient;
+
+  @SuppressWarnings("unused")
+  public static Stream<Arguments> provideTestCases() {
+    return Stream.of(
+        Arguments.of(
+            "Deployment distribution",
+            (Consumer<ZeebeClient>)
+                (client) -> { // given
+                },
+            (Function<ZeebeClient, CompletableFuture<?>>)
+                (client) -> { // when - network partition is set up
+                  final var process =
+                      Bpmn.createExecutableProcess("process").startEvent().endEvent().done();
+                  client.newDeployCommand().addProcessModel(process, "process.bpmn").send().join();
+                  return null;
+                },
+            (BiConsumer<ZeebeClient, CompletableFuture<?>>)
+                (client, future) -> { // then - after network partition
+                  final var topology = client.newTopologyRequest().send().join();
+
+                  final var partitions =
+                      IntStream.range(1, topology.getPartitionsCount() + 1)
+                          .boxed()
+                          .collect(Collectors.toSet());
+
+                  Awaitility.await("should be able to create instances on all partitions")
+                      .ignoreExceptions()
+                      .atMost(Duration.ofMinutes(1))
+                      .until(
+                          () -> {
+                            final var processInstanceEvent =
+                                client
+                                    .newCreateInstanceCommand()
+                                    .bpmnProcessId("process")
+                                    .latestVersion()
+                                    .send()
+                                    .join();
+
+                            return Protocol.decodePartitionId(
+                                processInstanceEvent.getProcessInstanceKey());
+                          },
+                          (partitionId) -> {
+                            LOGGER.info("Instance created on partition: {}", partitionId);
+                            partitions.remove(partitionId);
+                            return partitions.isEmpty();
+                          });
+                }),
+        Arguments.of(
+            "Message correlation",
+            (Consumer<ZeebeClient>)
+                (client) -> { // given
+                  final var process =
+                      Bpmn.createExecutableProcess("process")
+                          .startEvent()
+                          .intermediateCatchEvent()
+                          .message(msg -> msg.name("msg").zeebeCorrelationKeyExpression("key"))
+                          .endEvent()
+                          .done();
+                  client.newDeployCommand().addProcessModel(process, "process.bpmn").send().join();
+                  // make sure that the message is correlated to correct partition
```

Comment:
:question: Why is it important to which partition the message is correlated to?

Reply:
We partition away the partition three, and if the partition exist the message can't be correlated to the process instance BUT if the partition is taken down again the message should be correlated.